jpa/eclipselink.jpa.test.jse/src/it/java/org/eclipse/persistence/jpa/test/sequence/TestIdentityGeneration.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2022 IBM Corporation and/or affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+// Contributors:
+//     IBM - Bug 579409: Add support for accurate IDENTITY generation when the database contains separate TRIGGER objects
+package org.eclipse.persistence.jpa.test.sequence;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+
+import org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl;
+import org.eclipse.persistence.jpa.test.framework.DDLGen;
+import org.eclipse.persistence.jpa.test.framework.Emf;
+import org.eclipse.persistence.jpa.test.framework.EmfRunner;
+import org.eclipse.persistence.jpa.test.framework.Property;
+import org.eclipse.persistence.jpa.test.sequence.model.Coffee;
+import org.eclipse.persistence.jpa.test.sequence.model.Tea;
+import org.eclipse.persistence.jpa.test.sequence.model.TeaShop;
+import org.eclipse.persistence.platform.database.DatabasePlatform;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(EmfRunner.class)
+public class TestIdentityGeneration {
```

Comment:
Maybe tests there should be limited to the platforms with `this.supportsReturnGeneratedKeys == true` .

Reply:
I have `org.eclipse.persistence.jpa.test.sequence.TestIdentityGeneration.testRefreshWithTriggers()` gated at the beginning with 
```java
    Assume.assumeTrue(platform.supportsReturnGeneratedKeys());
```

I don't have `org.eclipse.persistence.jpa.test.sequence.TestIdentityGeneration.testPersistWithSecondaryTables()` gated with any check because there isn't anything inherently platform specific for that test. It should run successfully against any database, regardless of their support for "RETURN_GENERATED_KEYS". The test was included to exercise the code paths I changed and test for failures.